### ISSUE TITLE
[Bug]: Impossible to modify a Calendar Event after one Attendee has moved this event to the trash bin

### ISSUE THREAD

**Issue by fabieins** [nextcloud/server#32573]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on Github _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

Generate a Calendar event with Attendees on the same nextcloud instance.
After one of the Attendees moves this Event to the trash bin, the owner can't update the event.

after clicking the "Update"-Button there is a HTTP 400 Error with this Response:
`<?xml version="1.0" encoding="utf-8"?>
<d:error xmlns:d="DAV:" xmlns:s="http://sabredav.org/ns">
  <s:exception>Sabre\DAV\Exception\BadRequest</s:exception>
  <s:message>Deleted calendar object with uid already exists in this calendar collection.</s:message>
</d:error>`

When the event is deleted permanently from the trash bin by the attendee, the owner can modify the event again 

### Steps to reproduce

1. Create a Calendar Event with Attendees on the same nextcloud server
2. Wait until one Attendee moves this event to the trash bin
3. Try to update this Event.

### Expected behavior

Being able to update a Calendar Event as owner also when some of the Attendees moves this Event to the trash bin

### Installation method

Manual installation

### Operating system

Debian/Ubuntu

### PHP engine version

PHP 7.4

### Web server

Apache (supported)

### Database engine version

MariaDB

### Are you using the Nextcloud Server Encryption module?

Encryption is Disabled

### What user-backends are you using?

- [ ] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [X] SSO - SAML
- [ ] Other

### Configuration report

```shell
{
    "system": {
        "memcache.local": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "port": 6379
        },
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "***REMOVED SENSITIVE VALUE***"
        ],
        "htaccess.RewriteBase": "\/",
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "22.2.8.1",
        "overwrite.cli.url": "***REMOVED SENSITIVE VALUE***",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "updater.release.channel": "stable",
        "maintenance": false,
        "theme": "",
        "loglevel": "3",
        "mail_smtpmode": "smtp",
        "mail_smtpsecure": "tls",
        "mail_sendmailmode": "smtp",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "587",
        "mail_smtpauthtype": "PLAIN",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "app_install_overwrite": [
            "whiteboard",
            "bookmarks_fulltextsearch"
        ],
        "default_phone_region": "***REMOVED SENSITIVE VALUE***"
    }
}
```

### List of activated Apps

```shell
Enabled:
  - accessibility: 1.8.0
  - activity: 2.15.0
  - appointments: 1.12.4
  - apporder: 0.15.0
  - bookmarks: 10.3.1
  - breezedark: 22.3.1
  - bruteforcesettings: 2.4.0
  - calendar: 3.3.1
  - circles: 22.3.1
  - cloud_federation_api: 1.5.0
  - comments: 1.12.0
  - contacts: 4.1.1
  - contactsinteraction: 1.3.0
  - dashboard: 7.2.0
  - dav: 1.20.0
  - deck: 1.5.6
  - drawio: 1.0.2
  - extract: 1.3.5
  - federatedfilesharing: 1.12.0
  - federation: 1.12.0
  - files: 1.17.0
  - files_fulltextsearch: 22.0.2
  - files_fulltextsearch_tesseract: 22.0.0
  - files_markdown: 2.3.6
  - files_mindmap: 0.0.26
  - files_pdfviewer: 2.3.1
  - files_rightclick: 1.1.0
  - files_sharing: 1.14.0
  - files_texteditor: 2.14.0
  - files_trashbin: 1.12.0
  - files_versions: 1.15.0
  - files_videoplayer: 1.11.0
  - firstrunwizard: 2.11.0
  - forms: 2.5.0
  - fulltextsearch: 22.0.1
  - fulltextsearch_elasticsearch: 22.0.1
  - groupfolders: 10.0.2
  - integration_whiteboard: 0.0.14
  - logreader: 2.7.0
  - lookup_server_connector: 1.10.0
  - nextcloud_announcements: 1.11.0
  - notes: 4.3.1
  - notifications: 2.10.1
  - oauth2: 1.10.0
  - onlyoffice: 7.3.4
  - password_policy: 1.12.0
  - photos: 1.4.0
  - polls: 3.6.1
  - privacy: 1.6.0
  - provisioning_api: 1.12.0
  - recommendations: 1.1.0
  - serverinfo: 1.12.0
  - settings: 1.4.0
  - sharebymail: 1.12.0
  - spreed: 12.2.5
  - support: 1.5.0
  - survey_client: 1.10.0
  - systemtags: 1.12.0
  - tasks: 0.14.4
  - text: 3.3.1
  - theming: 1.13.0
  - twofactor_backupcodes: 1.11.0
  - updatenotification: 1.12.0
  - user_saml: 5.0.0
  - user_status: 1.2.0
  - viewer: 1.6.0
  - weather_status: 1.2.0
  - whiteboard: 0.0.3
  - workflowengine: 2.4.0
Disabled:
  - admin_audit
  - encryption
  - files_external
  - user_ldap
```

### Nextcloud Signing status

```shell
No errors have been found.
```

### Additional info

Could be related to https://github.com/nextcloud/server/issues/30096

For the owner of the Event there is no way to see, which Attendee has moved this Event to the trash bin.

**Comment by szaimen** [comment]
Hi, please update to 24.0.9 or better 25.0.3 and report back if it fixes the issue. Thank you!

My goal is to add a label like e.g. 25-feedback to this ticket of an up-to-date major Nextcloud version where the bug could be reproduced. However this is not going to work without your help. So thanks for all your effort!

If you don't manage to reproduce the issue in time and the issue gets closed but you can reproduce the issue afterwards, feel free to create a new bug report with up-to-date information by following this link: https://github.com/nextcloud/server/issues/new?assignees=&labels=bug%2C0.+Needs+triage&template=BUG_REPORT.yml&title=%5BBug%5D%3A+

**Comment by fabieins** [comment]
This issue still exists in 25.0.6 but seems to be fixed since 26.0.1 and calender-app 4.3.3
